### PR TITLE
Bob/4044 fix error focus

### LINKED ISSUE #4044: Return keyboard focus to form errors on person form
## Area or Component
Person form (add/edit person or self-registration)

## Quick Summary
The form doesn't return keyboard focus to the offending element in case of an erroring submission. Instead it scrolls to the top of the page. It should scroll to the first error (ideally after focusing the keyboard on the toast, but that's a separate problem)

### How to Replicate
1. Navigate to https://dev.simplereport.gov/app/register/E8Q2P (or another self-registration form in a lower environment)
2. Submit partial data, or invalid data (partial phone number, for instance)
3. Submit the form
4. Notice keyboard focus doesn't go back to the top error

## Accessibility Violation
https://dequeuniversity.com/class/forms/form-validation/focus-first-error

## Action Requested
Send focus to the first error on form submission.

If not already the case, we should also make sure to:
- Set invalid fields to aria-invalid="true".
- Ensure error messages are associated with form fields using aria-describedby so that screen reader users know how to fix the problem.

### PR DESCRIPTION
**Pull request by fzhao99**
# FRONTEND PULL REQUEST

## Related Issue
Fixes #4044 

## Changes Proposed

- Change the focus logic on submission to refocus based on the form schema
- Handle cases where there might be multiple entries in the form schema (phone)

## Testing

- Navigate to the patient self reg page on test: https://test.simplereport.gov/app/register/BBURF
- Submit incomplete registration info and verify that focus resets as expected
 
## Screenshots / Demos

https://user-images.githubusercontent.com/29645040/181042876-6ddc16d5-5f9d-4296-84b9-049511692837.mov

## Checklist for Author and Reviewer
- [x] Verify focus resets as expected
- [ ] Double check the implementation of the focus reset logic and make any suggestions for improvement.

### Security
- [ ] Changes with security implications have been approved by a security engineer (changes to  authentication, encryption, handling of PII, etc.)
- [ ] Any dependencies introduced have been vetted and discussed

### CODE REVIEW

**Review by emmastephenson — APPROVED** [review]
Just tested in `test` and it looks great, thanks for chasing down the phone number/email issue Bob!

**Comment by fzhao99** [comment]
@emmastephenson this is ready for review now! Ty to @georgehager for figuring out the testing issue I was running into 😊

File: frontend/src/app/patients/Components/ManageEmails.tsx [hunk 1]
```diff
@@ -123,7 +123,8 @@ const ManageEmails: React.FC<Props> = ({
           <div className="display-flex">
             <TextInput
               name={`email-${idx}`}
-              className="flex-fill"
+              idString={`email-${idx}`}
+              className="flex-fill emailFormElement"
```
Comment (emyl3) [inline frontend/src/app/patients/Components/ManageEmails.tsx:127]: Curious why we need this class? 🤔

File: frontend/src/app/patients/Components/ManagePhoneNumbers.tsx [hunk 2]
```diff
@@ -192,7 +202,8 @@ const ManagePhoneNumbers: React.FC<Props> = ({
         <div key={idx}>
           <div className="display-flex">
             <Input
-              className="flex-fill"
+              idString={`phoneInput-${idx}`}
+              className={`flex-fill phoneNumberFormElement`}
```
Comment (emyl3) [inline frontend/src/app/patients/Components/ManagePhoneNumbers.tsx:206]: Same here... Do we need the `phoneNumberFormElement` class?

**Review by emyl3 — APPROVED** [review]
Couple of very minor things I noticed. This looks great, Bob, and worked as intended! 😸

File: frontend/src/app/patients/Components/ManageEmails.tsx (hunk 1 above)
Comment (fzhao99) [reply frontend/src/app/patients/Components/ManageEmails.tsx:127]: We’re using the classes here to group all the potential error messages in the case of email / phone so we can grab and loop through them when applicable. Line 296 of the PersonForm.tsx file for reference!

Comment (emyl3) [reply frontend/src/app/patients/Components/ManageEmails.tsx:127]: Gotcha! Thanks for pointing me to that!

File: frontend/src/app/patients/Components/ManagePhoneNumbers.tsx (hunk 2 above)
Comment (emyl3) [reply frontend/src/app/patients/Components/ManagePhoneNumbers.tsx:206]: please ignore [and see above :) ](https://github.com/CDCgov/prime-simplereport/pull/4065#discussion_r937025620)

**Review by georgehager — APPROVED** [review]
Looks good! Glad we got the tests working 🎉

**Pull request merged**